### PR TITLE
refactor: use `data-url` crate to process data URLs in lsp & file_fetcher

### PR DESCRIPTION
**Pull request by satyarohith**
Closes https://github.com/denoland/deno/issues/10118

### CODE REVIEW

File: cli/lsp/urls.rs [hunk 1]
```diff
@@ -96,7 +87,11 @@ impl LspUrlMap {
         specifier.clone()
       } else {
         let specifier_str = if specifier.scheme() == "data" {
-          let media_type = data_url_media_type(specifier);
+          let data_url = DataUrl::process(specifier.as_str())
+            .map_err(|e| type_error(format!("{:?}", e)))?;
```
Comment (satyarohith) [inline cli/lsp/urls.rs:91]: Any suggestions on what error type would be appropriate here?

Comment (piscisaureus) [reply cli/lsp/urls.rs:91]: `URIError`

Comment (satyarohith) [reply cli/lsp/urls.rs]: Thanks! :)

File: cli/file_fetcher.rs [hunk 2]
```diff
@@ -762,39 +739,6 @@ mod tests {
     assert_eq!(file.source, expected);
   }
 
-  #[test]
-  fn test_get_source_from_data_url() {
```
Comment (piscisaureus) [inline cli/file_fetcher.rs:766]: Is there a reason to remove all these tests?
Are there any tests left that check that data URLs work properly?

Comment (satyarohith) [reply cli/file_fetcher.rs:766]: Yes. The removed tests check if data URLs are correctly processed. Now that we use the `data-url` crate, we no longer need them. And as far as the integration is concerned, the following two cases cover it.
https://github.com/denoland/deno/blob/f64630b8c152d9fb976c0ecdaa8abfff670932d8/cli/lsp/urls.rs#L188 and https://github.com/denoland/deno/blob/53414b4d3e6ef35e39ba8dacf8b406962ac99980/cli/file_fetcher.rs#L1001

**Review by piscisaureus — APPROVED** [review]
Thanks @satyarohith, LGTM.

**Pull request merged**